tests/components/jellyfin/test_media_source.py

```diff
@@ -0,0 +1,429 @@
+"""Tests for the Jellyfin media_source platform."""
+from unittest.mock import MagicMock
+
+import pytest
+
+from homeassistant.components.jellyfin.const import DOMAIN
+from homeassistant.components.media_player import MediaClass
+from homeassistant.components.media_player.errors import BrowseError
+from homeassistant.components.media_source import (
+    DOMAIN as MEDIA_SOURCE_DOMAIN,
+    URI_SCHEME,
+    async_browse_media,
+    async_resolve_media,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from . import load_json_fixture
+
+from tests.common import MockConfigEntry
+
+
+async def test_resolve(
+    hass: HomeAssistant,
+    mock_client: MagicMock,
+    init_integration: MockConfigEntry,
+    mock_jellyfin: MagicMock,
+    mock_api: MagicMock,
+) -> None:
+    """Test resolving Jellyfin media items."""
+
+    # Test resolving a track
+    mock_api.get_item.side_effect = None
+    mock_api.get_item.return_value = load_json_fixture("track.json")
+
+    assert await async_setup_component(hass, MEDIA_SOURCE_DOMAIN, {})
+    play_media = await async_resolve_media(hass, f"{URI_SCHEME}{DOMAIN}/TRACK-UUID")
+
+    expected_url = "http://localhost/Audio/TRACK-UUID/universal?UserId=test-username,DeviceId=TEST-UUID,MaxStreamingBitrate=140000000"
+    assert play_media.mime_type == "audio/flac"
+    assert play_media.url == expected_url
+
+    # Test resolving a movie
+    mock_api.get_item.side_effect = None
+    mock_api.get_item.return_value = load_json_fixture("movie.json")
+
+    assert await async_setup_component(hass, MEDIA_SOURCE_DOMAIN, {})
+    play_media = await async_resolve_media(hass, f"{URI_SCHEME}{DOMAIN}/MOVIE-UUID")
+
+    expected_url = "http://localhost/Videos/MOVIE-UUID/stream?static=true,DeviceId=TEST-UUID,api_key=TEST-API-KEY"
+    assert play_media.mime_type == "video/mp4"
+    assert play_media.url == expected_url
+
+    # Test resolving an unsupported item
+    mock_api.get_item.side_effect = None
+    mock_api.get_item.return_value = load_json_fixture("unsupported-item.json")
+
+    assert await async_setup_component(hass, MEDIA_SOURCE_DOMAIN, {})
+    with pytest.raises(BrowseError):
+        await async_resolve_media(hass, f"{URI_SCHEME}{DOMAIN}/UNSUPPORTED-ITEM-UUID")
+
+
+async def test_root(
+    hass: HomeAssistant,
+    mock_client: MagicMock,
+    init_integration: MockConfigEntry,
+    mock_jellyfin: MagicMock,
+    mock_api: MagicMock,
+) -> None:
+    """Test browsing the Jellyfin root."""
+
+    assert await async_setup_component(hass, MEDIA_SOURCE_DOMAIN, {})
+    browse = await async_browse_media(hass, f"{URI_SCHEME}{DOMAIN}")
+
+    expected_child_item = {
```

Comment:
You might want to use syrup for this to have it auto create test responses for you.

Reply:
Didn't know that existed. Great suggestion.